### PR TITLE
add libpcap

### PR DESCRIPTION
**Pull request by demhademha**

### CODE REVIEW

File: makefiles/libpcap.mk [hunk 1]
```diff
@@ -0,0 +1,54 @@
+ifneq ($(PROCURSUS),1)
+$(error Use the main Makefile)
+endif
+
+SUBPROJECTS   += libpcap
```
Comment (m1stadev) [inline makefiles/libpcap.mk:5]: Formatting.

File: makefiles/libpcap.mk [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+ifneq ($(wildcard $(BUILD_WORK)/libpcap/.build_complete),)
+libpcap:
+	@echo "Using previously built libpcap."
+else
+libpcap: libpcap-setup
+	cd $(BUILD_WORK)/libpcap/build && cmake \
+		$(DEFAULT_CMAKE_FLAGS) \
+			-DINET6=ON \
+				-DPCAP_TYPE=bpf \
```
Comment (m1stadev) [inline makefiles/libpcap.mk:23]: Don't add in a tab on every line.

File: makefiles/libpcap.mk [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+					..
+	+$(MAKE) -C $(BUILD_WORK)/libpcap/build
+	+$(MAKE) -C $(BUILD_WORK)/libpcap/build install \
+		DESTDIR=$(BUILD_STAGE)/libpcap
+	touch $(BUILD_WORK)/libpcap/.build_complete
+endif
+
+libpcap-package: libpcap-stage
+	# libpcap.mk Package Structure
+	rm -rf $(BUILD_DIST)/{libpcap0.8,libpcap0.8-dev}
```
Comment (m1stadev) [inline makefiles/libpcap.mk:33]: Can you change this to `rm -rf $(BUILD_DIST)/{libpcap0.8{,-dev}`?

File: makefiles/libpcap.mk [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+endif
+
+SUBPROJECTS   += libpcap
+LIBPCAP_VERSION := 1.10.1
+DEB_LIBPCAP_V   ?= $(LIBPCAP_VERSION)
+
+libpcap-setup: setup
+	$(call GITHUB_ARCHIVE,the-tcpdump-group,libpcap,$(LIBPCAP_VERSION),libpcap-$(LIBPCAP_VERSION))
+	$(call EXTRACT_TAR,libpcap-$(LIBPCAP_VERSION).tar.gz,libpcap-libpcap-$(LIBPCAP_VERSION),libpcap)
+	mkdir -p $(BUILD_WORK)/libpcap/build
```
Comment (m1stadev) [inline makefiles/libpcap.mk:12]: Is a separate folder for building necessary?

**Review by m1stadev — CHANGES_REQUESTED** [review]

File: makefiles/libpcap.mk (hunk 4 above)
Comment (demhademha) [reply makefiles/libpcap.mk:12]: Well, every other cmake project does it, so I consider it a standard

File: makefiles/libpcap.mk [hunk 5]
```diff
@@ -0,0 +1,56 @@
+ifneq ($(PROCURSUS),1)
+$(error Use the main Makefile)
+endif
+
+SUBPROJECTS += libpcap
```
Comment (TheRealKeto) [inline makefiles/libpcap.mk:5]: Fix indent.

**Review by TheRealKeto — CHANGES_REQUESTED** [review]

File: makefiles/libpcap.mk (hunk 4 above)
Comment (m1stadev) [reply makefiles/libpcap.mk:12]: fair enough.

File: makefiles/libpcap.mk [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+	rm -rf $(BUILD_DIST)/{libpcap0.8{,-dev}
+	mkdir -p $(BUILD_DIST)/libpcap0.8/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/lib
+	mkdir -p $(BUILD_DIST)/libpcap0.8-dev/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/{lib,lib/pkgconfig}
+
+	
+	# libpcap.mk Prep libpcap0.8
+	cp -a $(BUILD_STAGE)/libpcap/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/{lib/libpcap.$(LIBPCAP_VERSION).dylib,lib/libpcap.A.dylib} $(BUILD_DIST)/libpcap0.8/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/lib/
+	
+	# libpcap.mk Prep libpcap0.8-dev
+	cp -a $(BUILD_STAGE)/libpcap/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/{bin,include,lib/libpcap.dylib,lib/libpcap.a,lib/pkgconfig/} $(BUILD_DIST)/libpcap0.8-dev/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/
```
[the remark this answers is not in the document]
Comment (demhademha) [reply makefiles/libpcap.mk:44]: I think my way is more logical

**Pull request merged**